Sources/Hummingbird/Configuration.swift

```diff
@@ -18,240 +18,238 @@ import Logging
+    /// will be passed through the HTTP channel
+    public let maxStreamedUploadSize: Int
+    /// Maximum size of data in flight while streaming request payloads before back pressure is applied.
+    public let maxStreamingBufferSize: Int
+    /// Defines the maximum length for the queue of pending connections
+    public let backlog: Int
+    /// Allows socket to be bound to an address that is already in use.
+    public let reuseAddress: Bool
+    /// Disables the Nagle algorithm for send coalescing.
+    public let tcpNoDelay: Bool
+    /// Pipelining ensures that only one http request is processed at one time
+    public let enableHttpPipelining: Bool
+    /// Idle state handler setup.
+    public let idleTimeoutConfiguration: IdleStateHandlerConfiguration?
+    #if canImport(Network)
+    /// TLS options for NIO Transport services
+    public let tlsOptions: TSTLSOptions
+    #endif
+
+    /// Number of threads to allocate in the application thread pool
+    public let threadPoolSize: Int
+    /// don't run the HTTP server
+    public let noHTTPServer: Bool
+    /// logging level
+    public let logLevel: Logger.Level
+
+    // MARK: Initialization
+
+    /// Initialize HBApplication configuration
+    ///
+    /// - Parameters:
+    ///   - address: Bind address for server
+    ///   - serverName: Server name to return in "server" header
+    ///   - maxUploadSize: Maximum upload size allowed for routes that don't stream the request payload
+    ///   - maxStreamedUploadSize: Maximum upload size allowed when streaming data
+    ///   - maxStreamingBufferSize: Maximum size of data in flight while streaming request payloads before back pressure is applied.
+    ///   - backlog: the maximum length for the queue of pending connections.  If a connection request arrives with the queue full,
+    ///         the client may receive an error with an indication of ECONNREFUSE
+    ///   - reuseAddress: Allows socket to be bound to an address that is already in use.
+    ///   - tcpNoDelay: Disables the Nagle algorithm for send coalescing.
+    ///   - enableHttpPipelining: Pipelining ensures that only one http request is processed at one time
+    ///   - threadPoolSize: Number of threads in application thread pool
+    ///   - logLevel: Logging level
+    ///   - noHTTPServer: Don't start up the HTTP server.
+    public init(
+        address: HBBindAddress = .hostname(),
+        serverName: String? = nil,
+        maxUploadSize: Int = 1 * 1024 * 1024,
+        maxStreamedUploadSize: Int = 4 * 1024 * 1024,
+        maxStreamingBufferSize: Int = 1 * 1024 * 1024,
+        backlog: Int = 256,
+        reuseAddress: Bool = true,
+        tcpNoDelay: Bool = false,
+        enableHttpPipelining: Bool = true,
+        idleTimeoutConfiguration: IdleStateHandlerConfiguration? = nil,
+        threadPoolSize: Int = 2,
+        logLevel: Logger.Level? = nil,
+        noHTTPServer: Bool = false
```

Comment:
I don't like the concept and naming of `noHTTPServer`. First of all because this leads to a double-negative `noHTTPServer = false` means that the HTTP server is enabled.

But I also feel that `noHTTPServer` _really_ meas that this entire HTTPServer configuration could just be `nil`

Reply:
We can probably ditch this given the way the Service code works. It was mainly there to run a Jobs client without an HTTP server. Instead someone can just run the Jobs service on its own.

Reply:
Let's